### PR TITLE
Implement django forms

### PR DESCRIPTION
**Pull request by mitchdawson1982**
- Implements a search form
- implements services for search and details to create a fully populated context for a search of details request.

### CODE REVIEW

File: home/service/search.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            form_data = self.form.cleaned_data
+        else:
+            form_data = {}
+        query = form_data.get("query", "")
+        sort = form_data.get("sort", "relevance")
+        domains = form_data.get("domains", [])
+        filter_value = [MultiSelectFilter(
+            "domains", domains)] if domains else []
+        page_for_search = str(int(page) - 1)
+        if sort == "ascending":
```
Comment (PriyaBasker23) [inline home/service/search.py:31]: Refactor to single line

File: templates/partial/filter.html [hunk 2]
```diff
@@ -17,40 +18,57 @@ <h2 class="govuk-heading-m">Selected filters</h2>
...
+              href="{% url 'home:search' %}{% query_string domains=None %}"
               id="clear_filter"
             >
                 Clear filter
             </a>
             </p>
             </div>
           </div>
-          {% if selected_domain %}
+          <!-- {% if selected_domain %}
```
Comment (PriyaBasker23) [inline templates/partial/filter.html:29]: Can we remove these lines

File: templates/partial/pagination.html [hunk 3]
```diff
@@ -1,7 +1,8 @@
+{% load future %}
 <nav class="govuk-pagination" role="navigation" aria-label="Pagination">
     {% if page_obj.has_previous %}
     <div class="govuk-pagination__prev">
-      <a class="govuk-link govuk-pagination__link" href="{% url 'home:pagination' page=page_obj.previous_page_number %}" rel="prev">
+      <a class="govuk-link govuk-pagination__link" href="{% url 'home:pagination' page=page_obj.previous_page_number %}{% query_string clear_label=None clear_filter=None value=None %}" rel="prev">
```
Comment (PriyaBasker23) [inline templates/partial/pagination.html:5]: What is value in this ?

File: templates/partial/filter.html [hunk 4]
```diff
@@ -17,40 +18,57 @@ <h2 class="govuk-heading-m">Selected filters</h2>
...
-                  {% for option in domainlist%}
+                  {% for domain_option in form.domains %}
                       <div class="govuk-checkboxes__item">
-                          <input form="searchform" class="govuk-checkboxes__input" id="domain-{{ option|lower }}" name="domain" type="checkbox" value="{{ option.value }}" {% if option.value in selected_domain.keys %}checked{% endif %}>
-                          <label class="govuk-label govuk-checkboxes__label" for="domain-{{ option.value|lower }}">{{ option.label }}</label>
+                        {{ domain_option.tag}}
+                        <label class="govuk-label govuk-checkboxes__label" for="{{ domain_option.id_for_label }}">
+                          {{domain_option.choice_label}}
+                        </label>
+                          <!-- <input form="searchform" class="govuk-checkboxes__input" id="domain-{{ option|lower }}" name="domain" type="checkbox" value="{{ value }}" {% if option.value in selected_domain.keys %}checked{% endif %}>
```
Comment (PriyaBasker23) [inline templates/partial/filter.html:54]: This can be removed aswell isnt it

Comment (mitchdawson1982) [reply templates/partial/filter.html:54]: removed

**Review by PriyaBasker23 — APPROVED** [review]
LGTM

File: templates/partial/pagination.html (hunk 3 above)
Comment (mitchdawson1982) [reply templates/partial/pagination.html]: resolved

Comment (mitchdawson1982) [reply templates/partial/pagination.html]: removed

**Review by PriyaBasker23 — APPROVED** [review]
LGTM

**Pull request merged**